drivers/hmis/app/models/hmis/hud/assessment.rb

```diff
@@ -7,21 +7,41 @@
 class Hmis::Hud::Assessment < Hmis::Hud::Base
   include ::HmisStructure::Assessment
   include ::Hmis::Hud::Shared
+  include ArelHelper
+
   self.table_name = :Assessment
   self.sequence_name = "public.\"#{table_name}_id_seq\""
 
   belongs_to :enrollment, **hmis_relation(:EnrollmentID, 'Enrollment')
+  belongs_to :client, **hmis_relation(:PersonalID, 'Client')
   belongs_to :user, **hmis_relation(:UserID, 'User'), inverse_of: :assessments
   has_one :assessment_detail, class_name: 'Hmis::Form::AssessmentDetail'
+  belongs_to :data_source, class_name: 'GrdaWarehouse::DataSource'
+  has_one :wip, class_name: 'Hmis::Wip', as: :source
 
   use_enum :assessment_types_enum_map, ::HUD.assessment_types
   use_enum :assessment_levels_enum_map, ::HUD.assessment_levels
   use_enum :prioritization_statuses_enum_map, ::HUD.prioritization_statuses
 
   attr_writer :skip_validations
+  attr_accessor :in_progress
 
   validates_with Hmis::Hud::Validators::AssessmentValidator
 
+  scope :viewable_by, ->(user) do
+    enrollment_ids = Hmis::Hud::Enrollment.viewable_by(user).pluck(:EnrollmentID)
+    vieawable_wip = wip_t[:enrollment_id].in(enrollment_ids)
+    viewable_enrollment = as_t[:EnrollmentID].in(enrollment_ids)
+
+    left_outer_joins(:wip).where(vieawable_wip.or(viewable_enrollment))
+  end
+
+  def enrollment
+    super || Hmis::Hud::Enrollment.find_by(enrollment_id: wip.enrollment_id)
```

Comment:
Along with the other suggested change, this should be
```suggestion
    super || Hmis::Hud::Enrollment.find(wip.enrollment_id)
```

In the `save_in_progress` method we are storing the enrollment's _database id_ on the WIP record.